core/arch/arm/kernel/thread_a64.S

```diff
@@ -183,6 +183,7 @@ FUNC thread_std_smc_entry , :
 	mov	x0, sp
 
 	/* Call the registered handler */
+	mov	x29, #0
```

Comment:
I would have added: `thread->regs.x[29] = 0` at the end of `init_regs()` in `core/arch/arm/kernel/thread.c` instead.